Jenkinsfile

```diff
@@ -26,15 +29,70 @@ def withGo(Closure body) {
     }
 }
 
-def runIntegrationTest(String description, String kubeprodArgs, String ginkgoArgs, Closure setup) {
+def waitForRollout(String namespace, int minutes) {
+    withEnv(["PATH+KTOOL=${tool 'kubectl'}"]) {
+        try {
+            timeout(time: minutes, unit: 'MINUTES') {
+                sh """
+set +x
+for deploy in \$(kubectl --namespace ${namespace} get deploy --output name)
+do
+  echo -n "\nWaiting for rollout of \${deploy} in ${namespace} namespace"
+  while ! \$(kubectl --namespace ${namespace} rollout status \${deploy} --watch=false | grep -q "successfully rolled out")
+  do
+    echo -n "."
+    sleep 3
+  done
+done
+"""
+            }
+        } catch (error) {
+            sh "kubectl --namespace ${namespace} get po,deploy,svc,ing"
+            throw error
+        }
+    }
+}
+
+def insertGlueRecords(String name, String nameServersJson, String ttl, String zone, String resourceGroup) {
+    withCredentials([azureServicePrincipal('jenkins-bkpr-owner-sp')]) {
+        container('az') {
+            sh "az login --service-principal -u \$AZURE_CLIENT_ID -p \$AZURE_CLIENT_SECRET -t \$AZURE_TENANT_ID"
+            sh "az account set -s \$AZURE_SUBSCRIPTION_ID"
+            for (ns in readJSON(text: nameServersJson)) {
```

Comment:
Minor: Probably makes sense to make the `nameServers` argument to this function be an array/list of strings and call `readJSON` in the caller(s).  So far 2 out of 2 give it to us as a json-serialised array, so I'm also ok with leaving that as the "native form" for now...